src/Compilers/CSharp/Portable/Lowering/LocalRewriter_NullChecking.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        internal BoundStatement RewriteNullChecking(ImmutableArray<BoundStatement> statements, ImmutableArray<LocalSymbol> locals)
+        {
+            if (_factory.CurrentFunction.Parameters.Any(x => x is SourceParameterSymbolBase param
+                                                             && param.IsNullChecked))
+            {
+                return (BoundStatement)AddNullChecksToBody(statements, locals);
+            }
+            return _factory.Block(locals, statements);
+        }
+
+        private BoundNode AddNullChecksToBody(ImmutableArray<BoundStatement> bodyStatements, ImmutableArray<LocalSymbol> locals)
+        {
+            var statementList = ArrayBuilder<BoundStatement>.GetInstance();
+            foreach (ParameterSymbol x in _factory.TopLevelMethod.Parameters)
```

Comment:
>_factory.TopLevelMethod [](start = 42, length = 23)

We're using `TopLevelMethod` here but `CurrentFunction` above. Are the two always the same?